src/main/java/io/github/thebusybiscuit/slimefun4/api/events/MultiBlockCraftEvent.java

```diff
@@ -0,0 +1,85 @@
+package io.github.thebusybiscuit.slimefun4.api.events;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+import org.bukkit.inventory.ItemStack;
+
+import io.github.thebusybiscuit.slimefun4.core.multiblocks.MultiBlockMachine;
+
+/**
+ * This {@link Event} is called when a {@link Player} crafts an item using a {@link MultiBlockMachine}.
+ * Unlike the {@link MultiBlockInteractEvent}, this event only fires if an output to a craft is expected.
+ * If this event is cancelled, ingredients will not be consumed and no output item results.
+ *
+ * @author char321
+ *
+ */
+public class MultiBlockCraftEvent extends PlayerEvent implements Cancellable {
```

Comment:
I'd rather call it MultiBlockUseEvent

Reply:
i feel like that would make it easily confusable with multiblockinteractevent